manager/node.go

```diff
@@ -72,6 +72,9 @@ func (m *VolumeManager) DiskUpdate(name string, updateDisks []types.DiskSpec) (*
 			}
 		}
 		if !isInvalid {
+			if uDisk.StorageReserved < 0 || uDisk.StorageReserved > uDisk.StorageMaximum {
+				return nil, fmt.Errorf("Update disk on node %v error: The storageReserved setting of disk %v is not valid, should be positive and no more than storageMaximum", name, uDisk.Path)
```

Comment:
Reserved may be more than Maximum if the node is down and Maximum has been reset to 0. I think if it's valid when user input it, we should be fine. Though need to check if everything has been shortcutted if Reserved was set but Maximum wasn't.

Reply:
I have filtered out disks which maximum has been set to 0, so I just check storageReserved need to no more than Maximum here. Since user couldn't edit maximum from UI, do you mean we just need to check storageReserved should less than Maximum if Maximum is not set? But in tests, if storageReserved and Maximum both set, we don't need to valid them?

Reply:
Yeah, I see now you only validate if the disk is valid. That makes sense.